lib/TerraformDevKit/s3.rb

```diff
@@ -0,0 +1,43 @@
+require 'aws-sdk'
+
+Aws.use_bundled_cert!
+
+module TerraformDevKit
+  # Wrapper class around aws s3
+  class S3
+    def initialize(credentials, region)
+      @s3_client = Aws::S3::Client.new(
+        credentials: credentials,
+        region: region
+      )
+    end
+
+    def create_bucket(bucket_name)
+      @s3_client.create_bucket(
+        bucket: bucket_name
+      )
+    end
+
+    def delete_bucket(bucket_name)
+      empty_bucket(bucket_name)
+
+      @s3_client.delete_bucket(
+        bucket: bucket_name
+      )
+    end
+
+    def empty_bucket(bucket_name)
+      keys_to_delete = @s3_client
+                       .list_objects_v2(bucket: bucket_name)
```

Comment:
It might not be a issue, but it might be a good idea to implement support for pagination.

Reply:
We should only have one file in there so I didn't add pagination but if it's worth adding now I can add it.